### DISCUSSION TITLE
FFmpeg Videos not saving?

### DISCUSSION THREAD

**Discussion by AshhadQazi** [bonsai-rx/bonsai#1869] [Q&A]
Hello!

I have just been having some issues while trying to save recordings with Bonsai. I am using FFmpeg, and although just a simple workflow at the moment, the files are not saving. The codec I am using is: ffmpeg -i - -vcodec h264_nvenc -crf 23 -preset fast -b:v 5M. While the workflow is running, there is no issue, I can bring up the visualizer and the recording can run stable for a while. The issue is with the saving, no file saves. I have directed to the folder I want it to save in, ending in -.avi, yet nothing saves. During the recording however, I think that the FFmpeg is working, as processes such as the GPU, CPU, and RAM are being used and fluctuating. The file just doesn't save. Here are some images of what I see.
![image](https://github.com/bonsai-rx/bonsai/assets/170653199/fbd7d085-918d-48d2-be01-40e53a721f75)
This pops up after the recording starts.
![image](https://github.com/bonsai-rx/bonsai/assets/170653199/497f85bf-c9d9-4a64-a8ec-a534bcea3fd2)

I thought it might've been an improper installation of FFmpeg, so I reinstalled that and made sure to add it to the system variable PATH. I am pretty sure this shouldn't be the issue as I followed a recent tutorial. Although maybe someone has a different built FFmpeg that could work?

No errors pop up throughout the entire test, not even when ending the recording.

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/3c7a4517-1638-418a-909b-4c7682126158)

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/b73c6730-5d5c-4d79-bbde-947897d68940)

Using the original Videowriter works just fine.

Any help would be greatly appreciated!

**Comment by bruno-f-cruz** [comment]
Share your workflow, please.
Also, you have too many arguments in your `OutputArguments` string. if you want to call the full process consider using `StartProcess` operator, otherwise the operator already takes care of the `ffmpeg -i` portion for you. I suspect that what is going on is that ffmpeg allows duplicate arguments and runs with the most recent arguments in the string. By redifining "-i" you are probably deleting the reference to the original pipe that sends the raw video data to the ffmpeg process.

**Reply by AshhadQazi** [reply]
Sorry about that, here it is.
[FFmpegTest.zip](https://github.com/user-attachments/files/16058096/FFmpegTest.zip)

Does the Outputargument need to be filled in order for the process to work? Because when nothing is there, the error "Pipe is broken" appears.

**Reply by bruno-f-cruz** [reply]
When that happens, what error is thrown in the console?

Also, what happens if you simply write: `-vcodec h264_nvenc -crf 23 -preset fast -b:v 5M`?

**Reply by AshhadQazi** [reply]
Sorry about the late response, but with the two FFmpeg prompts, the one I originally used, and the one you asked for me to use, this error pops up:

Bonsai says that the "Pipe is Broken". Even when there is nothing in the OutputArguments string, same error.

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/111aef04-e83a-4f7b-b390-535ba3b524f7)

When using this string: ffmpeg -i - -vcodec h264_nvenc -crf 23 -preset fast -b:v 5M, no error pops up, but no video is being saved either. I tried running this on another station, and when running, the command prompt is showing a live framerate/processing feed. My guess is that FFmpeg isn't even able to gather the frames into video and save it? Could this be an issue with how the codec is written?
